test/tests/kernels/2d_diffusion/tests

```diff
@@ -70,4 +70,14 @@
     issues = '#11760'
     design = 'FEProblem.md'
   [../]
+
+  [./test_hypre_matrix]
+    type = 'Exodiff'
+    input = '2d_diffusion_test.i'
```

Comment:
Also, is there any way to tell that we are saving memory? How are we _actually_ testing this?